src/assets/toolkit/scripts/lib/component/slideshow1.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+export class Slideshow {
+  constructor (element, {
+    slideHolder = element.querySelector('.js-Slideshow-slides'),
+    slides = element.querySelectorAll('.js-Slideshow-slide'),
+    nextTrigger = element.querySelector('.js-Slideshow-next'),
+    prevTrigger = element.querySelector('.js-Slideshow-prev'),
+    currentCount = element.querySelector('.js-Slideshow-current'),
+    totalCount = element.querySelector('.js-Slideshow-total'),
+    classForward = 'slide-forward',
+    classForwardReset = 'slide-forwardReset',
+    classBack = 'slide-back'
+  } = {}) {
+
+    this.slides = Array.from(slides);
+    this.numSlides = this.slides.length;
+    this.counter = 0;
+    totalCount.innerHTML = this.numSlides;
+
+    Object.assign(this, {
+      element,
+      slideHolder,
+      nextTrigger,
+      prevTrigger,
+      currentCount,
+      totalCount,
+      classForward,
+      classForwardReset,
+      classBack
+    });
+
+    this.attachEvents();
+  }
+
+  updateCurrent () {
+    if (this.counter > 0) {
+      var currentSlide = (this.counter % this.numSlides);
+    } else {
+      var currentSlide = (this.counter % this.numSlides) ? ((this.counter % this.numSlides) + this.numSlides) : 0;
```

Comment:
I've tried for the last ten years and I still don't understand what the `%` operator does. 😓